webpack/assets/javascripts/react_app/components/hosts/storage/vmware/controller/disk/index.js

```diff
@@ -25,91 +25,84 @@ const Disk = ({
   storagePods,
   storagePodsStatus,
   storagePodsError,
-}) => {
-  const updateStoragePod = newValues => {
-    updateDisk('storagePod', newValues);
-    updateDisk('datastore', { target: { value: null } });
-  };
-  const updateDatastore = newValues => {
-    updateDisk('datastore', newValues);
-    updateDisk('storagePod', { target: { value: null } });
-  };
-
-  return (
-    <div className="disk-container">
-      <div className="form-group">
-        <label className="col-md-2 control-label">{__('Disk name')}</label>
-        <div className="col-md-4">{name}</div>
-        <div className="col-md-2">
-          {!vmExists && (
-            <Button className="close" onClick={removeDisk}>
-              <span aria-hidden="true">&times;</span>
-            </Button>
-          )}
-        </div>
+}) => (
+  <div className="disk-container">
+    <div className="form-group">
+      <label className="col-md-2 control-label">{__('Disk name')}</label>
+      <div className="col-md-4">{name}</div>
+      <div className="col-md-2">
+        {!vmExists && (
+          <Button className="close" onClick={removeDisk}>
+            <span aria-hidden="true">&times;</span>
+          </Button>
+        )}
       </div>
-      {!(datastore && datastore.length) && (
-        <Select
-          label={__('Storage Pod')}
-          value={storagePod}
-          disabled={vmExists}
-          onChange={newValues => updateStoragePod(newValues)}
-          options={storagePods}
-          allowClear
-          key="storagePodsSelect"
-          status={storagePodsStatus}
-          errorMessage={storagePodsError}
-          className="storage-pod"
-        />
-      )}
-      {!(storagePod && storagePod.length) && (
-        <Select
-          disabled={vmExists}
-          label={__('Data store')}
-          value={datastore}
-          onChange={newValues => updateDatastore(newValues)}
-          options={datastores}
-          allowClear
-          key="datastoresSelect"
-          status={datastoresStatus}
-          errorMessage={datastoresError}
-          className="datastore"
-        />
-      )}
-
+    </div>
+    {!(datastore && datastore.length) && (
       <Select
-        label={__('Disk Mode')}
-        value={mode}
+        label={__('Storage Pod')}
+        value={storagePod}
         disabled={vmExists}
-        onChange={newValues => updateDisk('mode', newValues)}
-        options={diskModeTypes}
+        onValueChange={newValue =>
+          updateDisk({ storagePod: newValue, datastore: '' })
```

Comment:
In my opinion, keeping this as the default `onChange` will make much more sense,
you will get the event object and manipulate it however you'll like, for example:
```
handleOnChange = e => {
 const storagePod = e.target.value;
 updateDisk({ storagePod, datastore: '' })
}
```

Reply:
I just wanted to be consistent, but I guess we can solve it in another mean :)